docs/en/stores/basic-usage.md

```diff
@@ -0,0 +1,471 @@
+
+```ts
+const createCommand = createCommandFactory<State>();
+
+const commandOne = createCommand<{ one: string }>(({ payload }) => {
+	return [];
+});
+
+const commandTwo = createCommand<{ two: string }>(({ payload }) => {
+	return [];
+});
+
+const process = createProcess<State, { one: string; two: string }>('example', [commandOne, commandTwo]);
+
+process(store)({ one: 'one', two: 'two' });
+```
+
+## Widgets and Stores
+
+There are two state containers available for widgets: `Container` and `StoreProvider`. These containers connect the application store with a widget.
+
+Note that the documentation in this section is intended to show how widgets and state are connected. For a more detailed look see the [Widgets]() reference guide.
+
+### Container
+
+A `Container` is a widget that fully encapsulates another widget. It connects the store to the widget with a `getProperties` function.
+
+> widget/User.ts
+
+```ts
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import { WidgetBase } from '@dojo/framework/widget-core/WidgetBase';
+
+import * as css from './styles/Hello.m.css';
+
+interface UserProperties {
+	name?: string;
+}
+
+export class User extends WidgetBase<UserProperties> {
+	protected render() {
+		const { name = 'stranger' } = this.properties;
+		return <h1 classes={[css.root]}>{`Hello, ${name}`}</h1>;
+	}
+}
+```
+
+> widget/User.container.ts
+
+```ts
+import { createStoreContainer } from '@dojo/framework/stores/StoreInjector';
+import { Container } from '@dojo/framework/widget-core/Container';
+import { State } from '../interfaces';
+import User from './User';
+
+const StoreContainer = createStoreContainer<State>();
+
+const UserContainer = StoreContainer(User, 'state', {
+	getProperties({ get, path }) {
+		const name = get(path('user', 'current', 'name'));
+
+		return { name };
+	}
+});
+```
+
+In this example `UserContainer` wraps `User` to display the current user's name. `createStoreContainer` is wrapper that is used to ensure the proper typing of `getProperties`. `getProperties` is responsible for getting to data from the store and creating properties for the widget.
+
+The most important thing to note about a `Container` is its properties are a 1:1 mapping to the widget it wraps. In other words, all properties that appear on the widget should be provided by the `Container`.
+
+### StoreProvider
+
+A `StoreProvider` is a widget that has its own `renderer` and connects to the store. It is always encapsulated in another widget because a it does not have properties defined.
+
+Using `User.ts` from the example above, we can use a `StoreProvider` to provide application state.
+
+> widget/User.provider.ts
+
+```ts
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { w } from '@dojo/framework/widget-core/d';
+import StoreProvider from '@dojo/framework/stores/StoreProvider';
+import { State } from '../../interfaces';
+import User from './User';
+
+export class UserProvider extends WidgetBase {
```

Comment:
Is it necessary to have a `UserProvider` instead of using the `StoreProvider` inline in which ever parent widget requires it? It will be easier to demonstrate leveraging values from the parent widget, I don't want people to jump to creating a separate widget for each provider.

Reply:
Yeah. This usage pattern was a carry-over from another project that was migrating from containers. I wish I could go on a rant about when someone should create clean widgets vs store-coupled widgets.